### ISSUE TITLE
Dimmer item check / compare current value error

### ISSUE THREAD

**Issue by jimtng** [boc-tothefuture/openhab-jruby#21]
This code failed:

```ruby
DimmerItem << DimmerItem < 30 ? 100 : 0
```

I want to set DimmerItem to 100 if its current value is less than 30, or to 0 otherwise.

It failed due to the `<` operator.
```
Caused by: org.jruby.exceptions.NoMethodError: (NoMethodError) undefined method `<' for #<Java::OrgOpenhabCoreLibraryItems::DimmerItem:0x5e64d4b7>
```

Is there a way to get the DimmerItem's current value/state?

**Comment by jimtng** [comment]
I found that I needed to enclose the right side of `<<` with parentheses.

This code also produced an error:

```ruby
DimmerItem << (DimmerItem.state < 30 ? 100 : 0)
```
Error:
```
Caused by: org.jruby.exceptions.TypeError: (TypeError) class org.jruby.RubyFixnum cannot be cast to class org.openhab.core.library.types.DecimalType (org.jruby.RubyFixnum is in unnamed module of loader org.eclipse.osgi.internal.loader.EquinoxClassLoader @6ffd08cc; org.openhab.core.library.types.DecimalType is in unnamed module of loader org.eclipse.osgi.internal.loader.EquinoxClassLoader @e2b9fc2)
	at org.jruby.RubyComparable.<(org/jruby/RubyComparable.java:207) ~[?:?]
```

This seems to work:
```ruby
DimmerItem << (DimmerItem.state.intValue < 30 ? 100 : 0)
```

Could we make this code to work?
```ruby
DimmerItem << (DimmerItem < 30 ? 100 : 0)
```

**Comment by boc-tothefuture** [comment]
> Could we make this code to work?
> 
> DimmerItem << (DimmerItem < 30 ? 100 : 0)

Yes and we should.. I will need to refactor the DimmerItem has a Ruby Numeric, which is what it should be anyway.

**Comment by jimtng** [comment]
Fixed by #108
